packages/jaeger-ui/src/utils/tracking/index.tsx

```diff
@@ -45,6 +45,13 @@ const gaID = _get(config, 'tracking.gaID');
 export const isGaEnabled = isTest || isDebugMode || (isProd && Boolean(gaID));
 const isErrorsEnabled = isDebugMode || (isGaEnabled && Boolean(_get(config, 'tracking.trackErrors')));
 
+const cookieToDimension = _get(config, 'tracking.cookieToDimension');
+if (cookieToDimension) {
+  const match = ` ${document.cookie}`.match(new RegExp(`[; ]${cookieToDimension.cookie}=([^\\s;]*)`));
```

Comment:
- what is the meaning of the leading space?
- are you forced to use regexp because there is no structured API to access cookies?

Reply:
1) Leading space because the regex looks for `;` and/or ` ` as delimiter characters between cookies. and/or because the ` ` is optional in the spec. The leading space is necessary if the target cookie is the first cookie in the `document.cookie` string.
2) Correct. There are npm libraries for this (of course) but a library for a simple RegEx is overkill, and libraries may make certain assumptions about the cookies that aren't relevant here, meanwhile it's easy to know what our own RegEx is precisely doing.